src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -522,20 +523,22 @@ private bool ApplicableOperatorIs(MethodSymbol candidate, CSharpSyntaxNode node,
             BoundExpression operand,
             TypeSymbol operandType,
             bool hasErrors,
-            DiagnosticBag diagnostics)
+            DiagnosticBag diagnostics,
+            bool wasSwitch = false)
         {
             bool wasExpression;
-            return BindConstantPattern(node, operand, operandType, node.Expression, hasErrors, diagnostics, out wasExpression);
+            return BindConstantPattern(node, operand, operandType, node.Expression, hasErrors, diagnostics, out wasExpression, wasSwitch);
         }
 
-        private BoundPattern BindConstantPattern(
+        internal BoundPattern BindConstantPattern(
             CSharpSyntaxNode node,
             BoundExpression left,
             TypeSymbol leftType,
             ExpressionSyntax right,
             bool hasErrors,
             DiagnosticBag diagnostics,
-            out bool wasExpression)
+            out bool wasExpression,
+            bool wasSwitch)
```

Comment:
It is needed in order to distinguish the way constant patterns are type checked in switch versus all other pattern contexts. There is a PROTOTYPE comment saying where it will be used. I'm laying the groundwork for it here, since I'm sharing the functions that bind the constants now.